### ISSUE TITLE
Hyperparameter setting

### ISSUE THREAD

**Issue by YvoGao** [slei109/TART#1]
Hello, author! Can the parameter be set using tart.sh? The data I reproduced were much more skewed than those in the paper, especially at HuffPost. The 1-shot of 20newsgroup is similar, and the 5-shot is also slightly off. Other data sets also have some biases

10-shot
0.6716720322668552,20newsgroup,0.98,,[],False,,2,../data/20news.json,20newsgroup,0.1,bilstm,False,1,0.001,0.001,,train,7,8,5,10,False,reuters_False_data.json,20,,25,result/5-way_1-shot_20newsgroup_result.csv,False,3,1,,1000,100,1000,,100,5,../pretrain_wordvec/wiki.en.vec,../pretrain_wordvec

5-shot
0.7930320369005204,20newsgroup,0.98,,[],False,,2,../data/20news.json,20newsgroup,0.1,bilstm,False,1,0.001,0.001,,train,7,8,5,10,False,reuters_False_data.json,20,,25,result/5-way_5-shot_20newsgroup_result.csv,False,3,5,,1000,100,1000,,100,5,../pretrain_wordvec/wiki.en.vec,../pretrain_wordvec

1-shot
0.39820801581442355,0.08724249734104635,huffpost,0.98,,[],False,,2,../data/huffpost.json,huffpost,0.1,bilstm,False,1,0.001,0.001,,train,16,20,5,10,False,reuters_False_data.json,20,,25,result/5-way_1-shot_huffpost_result.csv,False,3,1,,1000,100,1000,,100,5,../pretrain_wordvec/wiki.en.vec,../pretrain_wordvec
5-shot
0.5700000269412995,0.07295171397582322,huffpost,0.98,,[],False,,2,../data/huffpost.json,huffpost,0.1,bilstm,False,1,0.001,0.001,,train,16,20,5,10,False,reuters_False_data.json,20,,25,result/5-way_5-shot_huffpost_result.csv,False,3,5,,1000,100,1000,,100,5,../pretrain_wordvec/wiki.en.vec,../pretrain_wordvec

**Comment by slei109** [comment]
If your question is whether you can modify the parameters in the tart.sh, then the answer is yes, you can modify the training parameters in the tart.sh.
Did you complete the whole training process? From the results, it seems that the model has not converged. The training process adopts an early stop strategy, so you need to wait until the training is complete before you can test the model.

**Comment by YvoGao** [comment]
Thank you for you reply. Yeah， I finish my training process. I find the lr is 1e-4 in paper, but is 1e-3 in tart.sh, I try to adopt it. But HuffPost also has a gap of about 8%  between paper results and mine.
huffpost,5,5,0.6065440267622471,0.07793038303663903,huffpost,0.98,,[],False,,0,../data/huffpost.json,huffpost,0.1,bilstm,False,1,0.0001,0.0001,,train,16,20,5,10,False,reuters_False_data.json,20,,25,result/result.csv,False,2023,5,,1000,100,1000,,100,5,../pretrain_wordvec/wiki.en.vec,../pretrain_wordvec
huffpost,5,5,0.622424027711153,0.07208093032779118,huffpost,0.98,,[],False,,3,../data/huffpost.json,huffpost,0.1,bilstm,False,1,0.0001,0.0001,,train,16,20,5,10,False,reuters_False_data.json,20,,25,result/result.csv,False,23,5,,1000,100,1000,,100,5,../pretrain_wordvec/wiki.en.vec,../pretrain_wordvec
huffpost,5,5,0.601120027512312,0.07660408752546764,huffpost,0.98,,[],False,,2,../data/huffpost.json,huffpost,0.1,bilstm,False,1,0.0001,0.0001,,train,16,20,5,10,False,reuters_False_data.json,20,,25,result/result.csv,False,3,5,,1000,100,1000,,100,5,../pretrain_wordvec/wiki.en.vec,../pretrain_wordvec
huffpost,5,1,0.4164080185294151,0.08415848390306922,huffpost,0.98,,[],False,,3,../data/huffpost.json,huffpost,0.1,bilstm,False,1,0.0001,0.0001,,train,16,20,5,10,False,reuters_False_data.json,20,,25,result/result.csv,False,23,1,,1000,100,1000,,100,5,../pretrain_wordvec/wiki.en.vec,../pretrain_wordvec
huffpost,5,1,0.4134400177299976,0.09202907724314832,huffpost,0.98,,[],False,,2,../data/huffpost.json,huffpost,0.1,bilstm,False,1,0.0001,0.0001,,train,16,20,5,10,False,reuters_False_data.json,20,,25,result/result.csv,False,3,1,,1000,100,1000,,100,5,../pretrain_wordvec/wiki.en.vec,../pretrain_wordvec
re
and my Hyperparameter following
python ../src/main.py \
    --cuda=$cuda \
    --way 5 \
    --shot 5 \
    --query 25 \
    --mode train \
    --dataset=$dataset \
    --data_path=$data_path \
    --n_train_class=$n_train_class \
    --n_val_class=$n_val_class \
    --n_test_class=$n_test_class \
    --embedding bilstm \
    --train_episodes 100 \
    --k 1 \
    --lr_g 1e-4 \
    --lr_d 1e-4 \
    --Comments "huffpost" \
    --patience 20 \
    --seed 3\


python ../src/main.py \
    --cuda=$cuda \
    --way 5 \
    --shot 1 \
    --query 25 \
    --mode train \
    --dataset=$dataset \
    --data_path=$data_path \
    --n_train_class=$n_train_class \
    --n_val_class=$n_val_class \
    --n_test_class=$n_test_class \
    --embedding bilstm \
    --train_episodes 100 \
    --k 1 \
    --lr_g 1e-4 \
    --lr_d 1e-4 \
    --Comments "huffpost" \
    --patience 20 \
    --seed 3\

**Comment by slei109** [comment]
May I see the outputs of the last several episodes? Once it finished training, you should see "End of training. Restore the best weights". Then it will start evaluate the performance with the testing dataset.

**Comment by YvoGao** [comment]
5-shot task
2023-08-07 12:48:45.603522, ep 28, val   acc: 0.6202 ± 0.0584, train stats G_grad: 0.2615, clf_grad: 0.1108
---------------ep:29 acc:0.6704000294208526-----------                          
2023-08-07 12:48:51.198213, ep 29, val   acc: 0.6303 ± 0.0526, train stats G_grad: 0.2363, clf_grad: 0.1077
---------------ep:30 acc:0.7841600334644317-----------                          
2023-08-07 12:48:57.627295, ep 30, train acc: 0.6763 ± 0.0754                   
2023-08-07 12:49:04.887994, ep 30, val   acc: 0.6158 ± 0.0597, train stats G_grad: 0.1636, clf_grad: 0.0858
2023-08-07 12:49:04.888127, End of training. Restore the best weights
2023-08-07 12:50:16.307526, test acc mean  0.6210, test std  0.0773

**Comment by slei109** [comment]
On our server, it typically takes about 60 epochs to converge on the huffpost dataset. If you keep the same batch size and the lr rate, it should produce the similar results. Did you utilize the same environment? I‘m not sure if a different version of pytorch would cause this problem.

**Comment by slei109** [comment]
We just uploaded the pre-trained weights for better reproduce. Hope this helps.

**Comment by yunlong-G** [comment]
> We just uploaded the pre-trained weights for better reproduce. Hope this helps.

Thank you for uploaded the pre-trained weights, where is the link

**Comment by slei109** [comment]
It's on the Readme page, next to dataset info.
